Task3/Task3_Nikolay_Varadinov/src/python/car_wash/libPyPool/utils.py

```diff
@@ -43,7 +43,7 @@ def __enter__(self):
             return ((self.pool, conn))
         except psycopg2.Error as error:
             logging.info(error.diag.message_primary)
-            return None
+            return ()
```

Comment:
I do not see a reason to have _Connection class. Also, your class is context manager class and you use it in another context manager. It is complicated. Please check my last comment in chat. Idea with lock is nice.